validator/sawtooth_validator/networking/interconnect.py

```diff
@@ -507,6 +508,17 @@ def setup(self, socket_type, complete_or_error_queue):
             self._socket.set(zmq.TCP_KEEPALIVE_IDLE, self._connection_timeout)
             self._socket.set(zmq.TCP_KEEPALIVE_INTVL, self._heartbeat_interval)
 
+            # Enable ZMQ support for IPv6
+            # For some reason, enabling this breaks binding to an interface
+            # name in Docker containers, so only enable it if an IPv6 address
+            # is explicitly specified.
+            try:
+                address = self._address.replace("[", "").replace("]", "")
+                ipaddress.IPv6Address(address)
```

Comment:
Can you just pass `self._address` to this?

Reply:
No, the ipaddress.IPv6Address class chokes on the bracket notation (but ZMQ parses it correctly).